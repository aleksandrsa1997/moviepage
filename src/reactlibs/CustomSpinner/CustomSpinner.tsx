import React from 'react';
import { CircularProgress } from '@material-ui/core';

export type TProps = {
    size?: number;
    value?: number;
    color?: 'inherit' | 'primary' | 'secondary';
    variant?: 'determinate' | 'indeterminate' | 'static';
    thickness?: number;
    disableShrink?: boolean;
};

const CustomSpinner: React.FC<TProps> = ({
    size = 40,
    color = 'primary',
    variant = 'determinate',
    thickness = 3.6
}) => (
    <CircularProgress
        size={size}
        color={color}
        variant={variant}
        thickness={thickness}
    />
);

export default CustomSpinner;