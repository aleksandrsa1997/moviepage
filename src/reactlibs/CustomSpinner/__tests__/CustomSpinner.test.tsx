import React from 'react';
import CustomSpinner from 'src/reactlibs/CustomSpinner';

describe('CustomSpinner snapshot', () => {
    it('should correct renderer', () => {
        const component = renderer(<CustomSpinner />);

        expect(component).toMatchSnapshot();
    });
});