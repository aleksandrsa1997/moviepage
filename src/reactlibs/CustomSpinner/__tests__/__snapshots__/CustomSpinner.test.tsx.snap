// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSpinner snapshot should correct renderer 1`] = `
<div
  aria-valuenow={0}
  className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
  role="progressbar"
  style={
    Object {
      "height": 40,
      "transform": "rotate(-90deg)",
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={
        Object {
          "strokeDasharray": "126.920",
          "strokeDashoffset": "126.920px",
        }
      }
    />
  </svg>
</div>
`;
